eth/stagedsync/stage_polygon_sync.go

```diff
@@ -378,15 +378,15 @@ func (s polygonSyncStageCheckpointStore) GetEntity(ctx context.Context, id uint6
 		return nil
 	})
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 	if r.err != nil {
-		return nil, r.err
+		return nil, false, r.err
```

Comment:
checkpointReader.Checkpoint returns "not found err"
let's map it to our new style here by checking if error.Is(notFound) then return `nil, false, nil`